### ISSUE TITLE
Copyright infringement on 6821.cpp (fixable)

### ISSUE THREAD

**Issue by iKarith** [linappleii/linapple#18]
The version of 6821.cpp we use from mame has an explicit license for non-commercial use only.  This runs afoul of the GNU GPL which prohibits such license terms.  We just need to refresh this using MAME's new license terms.  :)

**Comment by iKarith** [comment]
Also applies to AY8910.cpp.  These aren't the only Copyright problems, but they are two of them that are easily fixed because they come from MAME which got relicensed.

**Comment by maxolasersquad** [comment]
The 6821.cpp  on the WinApple project doesn't have this problem. I tried wholesale using using theirs as a drop-in replacement. Unsurprisingly, it didn't work.

**Comment by iKarith** [comment]
Yeah and the MAME version is unlikely to be a drop-in without modding.
These aren't the only Copyright concerns on the tree as a whole, they're
just the two for which the answer is obvious. I don't consider these two to
be the 100% highest priority, but the project will not be accepted into a
Linux distribution such as Debian until we fix it.  There's other code
clearly not written by Andrey in the tree whose author is unknown.
Fortunately it's stuff like a C++ genericized linked list template which is
probably available as public domain from any number of places—we may just
need to replace it with another one or roll one ourselves if we can't
identify the origin of this specific one.

Joseph

**Comment by ghedger** [comment]
Duly noted.

**Comment by rhaleblian** [comment]
I'm looking at upstream as i work on other issues, i will try to integrate from it gradually, hoping that we can reach a point where 6821.cpp becomes compatible.

**Comment by rhaleblian** [comment]
Oh, i made a promise-y thing here a while ago. Will look again.

**Comment by Michaelangel007** [comment]
Please keep us in the loop upstream for any other potential violations.  Thanks.
